### ISSUE TITLE
Improve documentation in case of use of external store

### ISSUE THREAD

**Issue by abenhamdine** [drcmda/react-contextual#13]
If I'm right, and according to my tests, two importants points are not clear in the docs :   

1°) if an external store is used (ie declared with createStore), the prop `to` is required in the subscribe component, and we must pass the store :

Eg, following won't work with an external store

```jsx
<Provider store={store}>
	<Subscribe select={(state) => state}>
		{(state: AppState) => {
			// TypeError : Cannot read property 'isLoggedIn' of undefined
			return state.login.isLoggedIn ? "Logged ?" : "Unlogged"
		}}
	</Subscribe>
</Provider>
```

whereas the following code works : 

```jsx
<Provider store={store}>
	<Subscribe to={store} select={(state) => state}>
		{(state: AppState) => {
			// ok, runs fine
			return state.login.isLoggedIn ? "Logged ?" : "Unlogged"
		}}
	</Subscribe>
</Provider>
```

**Note that I passed `store` to the Subscribe component, and not `ProviderContext` as described in docs here : https://github.com/drcmda/react-contextual/blob/master/API.md#subscribe-as-a-component, so I'm confused what is the correct way to suscribe to an external store. So what is `ProviderContext` ? Should I import the store in every component I want to suscribe ?**

2°) the external store must be passed to the provider with the `store` property :

eg :
```jsx
<Provider store={store}>
        <Subscribe to={store} select={(state) => state}>
            {props => (
                <div>
                    <h1>{props.count}</h1>
                    <button onClick={props.actions.up}>Up</button>
                    <button onClick={props.actions.down}>Down</button>
                </div>
            )}
        </Subscribe>
    </Provider>
```
 
whereas with 'internal' store, store properties are spread :
```jsx
<Provider {...store}>
        <Subscribe>
            {props => (
                <div>
                    <h1>{props.count}</h1>
                    <button onClick={props.actions.up}>Up</button>
                    <button onClick={props.actions.down}>Down</button>
                </div>
            )}
        </Subscribe>
    </Provider>
```
 

This should be emphasized in https://github.com/drcmda/react-contextual#external-store

**Comment by drcmda** [comment]
@abenhamdine good call! The internal store is more of a convenience thing, but i can see how it's confusing once you would use external stores or different context providers in general. 

> so I'm confused what is the correct way to suscribe to an external store. So what is ProviderContext ? Should I import the store in every component I want to suscribe ?

Correct, the internal store is an exception, everything else is referred to by reference:

```jsx
import store from '../assets/store'

<Subscribe to={store} />
```

It would be super helpful if you had any ideas how to word this properly. Going on a small vacation in a day and would be nice to have this in.

**Comment by abenhamdine** [comment]
Thx for you answer !
Actually, I have edited my initial post since I messed the code of the two examples (external vs internal store) and my comment about spread properties was related to internal store, because in the docs (https://github.com/drcmda/react-contextual#render-props), I see : 
```jsx
// see the spread here :
<Provider {...store}>
        <Subscribe>
            {props => (
                <div>
                    <h1>{props.count}</h1>
                    <button onClick={props.actions.up}>Up</button>
                    <button onClick={props.actions.down}>Down</button>
                </div>
            )}
        </Subscribe>
    </Provider>
```

Otherwise, one point is still not totally clear for me : IIUC ProviderContext (see https://github.com/drcmda/react-contextual/blob/master/API.md#subscribemapcontexttopropsanycomponent) references the internal store. And if there is an internal store, first arg of Subscribe can be omitted.
So is there a need for ProviderContext ?

> It would be super helpful if you had any ideas how to word this properly.

I can try a PR. It would be usefull for newcomers to have this kind of explanations !
